src/ring_theory/subsemiring.lean

```diff
@@ -831,3 +831,13 @@ instance [add_comm_monoid α] [module R' α] (S : subsemiring R') : module S α
 end subsemiring
 
 end actions
+
+-- while this definition is not about subrings, this is the earliest we have
+-- both ordered ring structures and submonoids available
+
+
+/--Submonoid of positive elements of an ordered semiring-/
+def pos_submonoid {R : Type*} [ordered_semiring R] [nontrivial R] : submonoid R :=
```

Comment:
I guess `R` should probably be explicit here too
```suggestion
def pos_submonoid (R : Type*) [ordered_semiring R] [nontrivial R] : submonoid R :=
```
The comment about the `simp` lemma above applies here too.